pkg/querier/queryrange/roundtrip.go

```diff
@@ -95,3 +98,75 @@ func NewLogFilterTripperware(
 		return next
 	}), nil
 }
+
+// NewMetricTripperware creates a new frontend tripperware responsible for handling metric queries
+func NewMetricTripperware(
+	cfg Config,
+	log log.Logger,
+	limits Limits,
+	codec queryrange.Codec,
+	extractor queryrange.Extractor,
+) (frontend.Tripperware, Stopper, error) {
+
+	queryRangeMiddleware := []queryrange.Middleware{queryrange.LimitsMiddleware(limits)}
+	if cfg.AlignQueriesWithStep {
+		queryRangeMiddleware = append(
+			queryRangeMiddleware,
+			queryrange.InstrumentMiddleware("step_align"),
+			queryrange.StepAlignMiddleware,
+		)
+	}
+
+	// SplitQueriesByDay is deprecated use SplitQueriesByInterval.
+	if cfg.SplitQueriesByDay {
+		level.Warn(log).Log("msg", "flag querier.split-queries-by-day (or config split_queries_by_day) is deprecated, use querier.split-queries-by-interval instead.")
+	}
+
+	queryRangeMiddleware = append(
+		queryRangeMiddleware,
+		queryrange.InstrumentMiddleware("split_by_interval"),
+		SplitByIntervalMiddleware(cfg, limits, codec),
+	)
+
+	var c cache.Cache
+	if cfg.CacheResults {
+		queryCacheMiddleware, cache, err := queryrange.NewResultsCacheMiddleware(
```

Comment:
I'm wondering how this will work since it also use a split duration and is not difficult to align.

see https://github.com/grafana/loki/blob/928d7a4c978793be350e58c13d67e5a7dddb88dd/vendor/github.com/cortexproject/cortex/pkg/querier/queryrange/results_cache.go#L108

Reply:
I don't think this will be important as tenants can't query other tenants' data. Thus, it should be just as reusable as the previous method but vary by tenant, no?

Reply:
nvm, it looks like the cache takes an interval config I'll have to account for. Good catch @cyriltovena